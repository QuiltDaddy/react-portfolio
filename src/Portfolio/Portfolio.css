.projectImg {
  width: 70%;
  margin-bottom: 4vw;
}

.projectText {
  font-size: calc(4px + 1.5vw);
  width: 70%;
  text-align: left;
  line-height: 4vw;
  font-family: 'Montserrat Alternates', sans-serif;
  margin-left: auto;
  margin-right: auto;
}

.projectHeader {
  display: flex;
  text-align: left;
  position: relative;
  left: 15%;
  width: 70%;
  margin-bottom: 1vw;
}

h2 {
  font-size: calc(5px + 2vw);
  font-family: 'Montserrat Alternates', sans-serif;
  text-align: left;
}

.icon {
  width: calc(12px + 2vw);
  height: calc(12px + 2vw);
  margin-left: 2px;
}

.HTML5 {
  height: calc(12px + 2vw);
  width: calc(13.5px + 2vw);
}

.CSS3 {
  height: calc(12px + 2vw);
  width: calc(9.5px + 2vw);
}

.linkContainer {
  text-align: left;
  position: relative;
  left: 15%;
  width: 70%;
  display: flex;
  justify-content: space-around;
  margin-top: .5vw;
  margin-bottom: .75vw;
}

.linkText {
  font-size: calc(4px + 1.5vw);
  line-height: 4vw;
  font-family: 'Montserrat Alternates', sans-serif;
  vertical-align: middle;
}

.repo {
  width: calc((4px + 1.5vw) * 2.85);
}

.liveApp {
  width: calc((4px + 1.5vw) * 4.62);
}

.iconSmall {
  width: calc(6px + 1.5vw);
  height: calc(6px + 1.5vw);
  vertical-align: middle;
  margin-left: .5vw;
}

@media screen and (max-width: 400px) {
  .icon {
    width: calc(5px + 2vw);
    height: calc(5px + 2vw);
    position: relative;
    top: 2px;
  }

  .HTML5 {
    height: calc(5px + 2vw);
    width: calc(6px + 2vw);
  }

  .CSS3 {
    height: calc(5px + 2vw);
    width: calc(4.2px + 2vw);
  }
}
