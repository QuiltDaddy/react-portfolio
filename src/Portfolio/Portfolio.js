import React, { Component } from 'react';
import projects from '../projects';
import portfolioIcons from '../pictures/portfolioIcons';
import './Portfolio.css';


class Portfolio extends Component {
  state = {
    projects
  };

  render() {
    const {projects} = this.state;
    
    const portfolio = projects.map((project, i) => {
      return (
       <div
         className='project'
         key={i}>
          <div className='projectHeader'>
            <h2>{project.name}</h2>
            {project.html ? (
              <img
                className='icon HTML5'
                src={portfolioIcons.html}
                alt='HTML5 icon' />
            ) : null}
            {project.css ? (
              <img
                className='icon CSS3'
                src={portfolioIcons.css}
                alt='CSS3 icon' />
            ) : null}
            {project.js ? (
              <img
                className='icon'
                src={portfolioIcons.js}
                alt='JS icon' />
            ) : null}
            {project.jquery ? (
              <img
                className='icon'
                src={portfolioIcons.jQuery}
                alt='jQuery icon' />
            ) : null}
            {project.react ? (
              <img
                className='icon'
                src={portfolioIcons.react}
                alt='react icon' />
            ) : null}
            {project.node ? (
              <img
                className='icon'
                src={portfolioIcons.node}
                alt='node icon' />
            ) : null}
            {project.postgres ? (
              <img
                className='icon'
                src={portfolioIcons.postgres}
                alt='postgres icon' />
            ) : null}
          </div>
          <p className='projectText'>
            {project.summary}
          </p>
          <div className='linkContainer'>
            <div>
              <label className='linkText repo'>
                Repo:
              </label>
              <a
                href={project.repoLink}
                target='_blank'
                rel='noopener noreferrer'>
                <img
                  className='iconSmall'
                  src={portfolioIcons.gitHub}
                  alt='GitHub icon' />
              </a>
            </div>
            <div>
              <label className='linkText liveApp'>
                Live App:
              </label>
            </div>
          </div>
          <a
            href={project.liveLink}
            target='_blank'
            rel='noopener noreferrer'>
            <img
              className='projectImg'
              src={project.screen}
              alt='A clickable screenshot of the application that takes you to the live website.' />
          </a>
        </div>
      );
    });

    return (
      <section className='portfolio'>
        <div className='anchor' id='portfolio'></div>
        <h1>Portfolio</h1>
        {portfolio}
      </section>
    );
  }
}

export default Portfolio;
