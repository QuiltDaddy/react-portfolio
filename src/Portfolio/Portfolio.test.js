import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Portfolio from './Portfolio';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Portfolio />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
const tree = renderer
  .create(<Portfolio />)
  .toJSON();
expect(tree).toMatchSnapshot();  
});
