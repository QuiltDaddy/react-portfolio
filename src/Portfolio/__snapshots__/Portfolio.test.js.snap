// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="portfolio"
>
  <div
    className="anchor"
    id="portfolio"
  />
  <h1>
    Portfolio
  </h1>
  <div
    className="project"
  >
    <div
      className="projectHeader"
    >
      <h2>
        Sudoku Solutions
      </h2>
      <img
        alt="HTML5 icon"
        className="icon HTML5"
        src="icon-HTML5.png"
      />
      <img
        alt="CSS3 icon"
        className="icon CSS3"
        src="icon-CSS3.png"
      />
      <img
        alt="JS icon"
        className="icon"
        src="icon-JS.png"
      />
      <img
        alt="jQuery icon"
        className="icon"
        src="icon-jQuery.png"
      />
    </div>
    <p
      className="projectText"
    >
      This application takes any given Sudoku and walks the user through the logic of solving it with detailed explanations at each step.  It also has a feature to generate a random new puzzle of a specified difficulty.  Accessible to A11y users, keyboard users, mouse users, and phone users.  Also included is a brute force solve method.
    </p>
    <div
      className="linkContainer"
    >
      <div>
        <label
          className="linkText repo"
        >
          Repo:
        </label>
        <a
          href="https://github.com/QuiltDaddy/Sudoku-Website"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub icon"
            className="iconSmall"
            src="icon-gitHub.png"
          />
        </a>
      </div>
      <div>
        <label
          className="linkText liveApp"
        >
          Live App:
        </label>
      </div>
    </div>
    <a
      href="https://quiltdaddy.github.io/Sudoku-Website/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="A clickable screenshot of the application that takes you to the live website."
        className="projectImg"
        src="sudoku-screen.png"
      />
    </a>
  </div>
  <div
    className="project"
  >
    <div
      className="projectHeader"
    >
      <h2>
        Yoga Quiz
      </h2>
      <img
        alt="HTML5 icon"
        className="icon HTML5"
        src="icon-HTML5.png"
      />
      <img
        alt="CSS3 icon"
        className="icon CSS3"
        src="icon-CSS3.png"
      />
      <img
        alt="JS icon"
        className="icon"
        src="icon-JS.png"
      />
      <img
        alt="react icon"
        className="icon"
        src="icon-react.png"
      />
    </div>
    <p
      className="projectText"
    >
      A program to demonstrate basic React website functionality and accessibility.  Accessible to A11y users, keyboard users, mouse users, and phone users.
    </p>
    <div
      className="linkContainer"
    >
      <div>
        <label
          className="linkText repo"
        >
          Repo:
        </label>
        <a
          href="https://github.com/QuiltDaddy/react-yoga-quiz/tree/master"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub icon"
            className="iconSmall"
            src="icon-gitHub.png"
          />
        </a>
      </div>
      <div>
        <label
          className="linkText liveApp"
        >
          Live App:
        </label>
      </div>
    </div>
    <a
      href="https://quiltdaddy.github.io/react-yoga-quiz/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="A clickable screenshot of the application that takes you to the live website."
        className="projectImg"
        src="yoga-screen.png"
      />
    </a>
  </div>
  <div
    className="project"
  >
    <div
      className="projectHeader"
    >
      <h2>
        Spouts on Routes
      </h2>
      <img
        alt="HTML5 icon"
        className="icon HTML5"
        src="icon-HTML5.png"
      />
      <img
        alt="CSS3 icon"
        className="icon CSS3"
        src="icon-CSS3.png"
      />
      <img
        alt="JS icon"
        className="icon"
        src="icon-JS.png"
      />
      <img
        alt="jQuery icon"
        className="icon"
        src="icon-jQuery.png"
      />
    </div>
    <p
      className="projectText"
    >
      This project integrates Open Brewery DB, Google Maps API, and Mapquest Geocoding API to provide the user with bicycle directions to their favorite brewery.  After searching their area for craft or large scale breweries the user provides the desired starting address and the bicycle optimized directions are displayed visually and in text format.
    </p>
    <div
      className="linkContainer"
    >
      <div>
        <label
          className="linkText repo"
        >
          Repo:
        </label>
        <a
          href="https://github.com/QuiltDaddy/spouts-on-routes"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub icon"
            className="iconSmall"
            src="icon-gitHub.png"
          />
        </a>
      </div>
      <div>
        <label
          className="linkText liveApp"
        >
          Live App:
        </label>
      </div>
    </div>
    <a
      href="https://quiltdaddy.github.io/spouts-on-routes/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="A clickable screenshot of the application that takes you to the live website."
        className="projectImg"
        src="spouts-screen.png"
      />
    </a>
  </div>
</section>
`;
